pkg/scale/decode.go

```diff
@@ -510,56 +573,56 @@ func (ds *decodeState) decodeFixedWidthInt(dstv reflect.Value) (err error) {
 		var b byte
 		b, err = ds.ReadByte()
 		if err != nil {
-			break
+			return
```

Comment:
question: why do you return after these errors here?  While in `decodeCustomPrimitive` 148 - 230 you break on errors.  I'm wondering why the difference?

Reply:
Sorry for not responding @edwardmack! It's because of line 632: 
```
dstv.Set(reflect.ValueOf(out))
```
I didn't want to `Set` on the error case, so I had to return.  Conversely, I could've put that line in every case. I dunno which is better.

Reply:
@timwu20 no problem, I'm just asking to better understand. So, at 229 you do want `dstv.Set(temp.Elem().Convert(inType))` set on error?

Reply:
No I don't.  if `ds.ReadByte` returns an error, I'm just exiting this `decodeFixedWidthInt`.  I had it as `break` before, but then it would panic cause it would hit line 632.